big-int/src/operators/equal_to.cpp

```diff
@@ -44,11 +44,74 @@ bool largeInt::operator==(const largeInt &z)
     }
     for (int i = 0; i < this->number.length(); i++)
     {
+        // iterates through all digits of both inputs
+        // if two digits are not the same, returns false
         if (this->number[i] != z.number[i])
         {
             return false;
         }
     }
     return true;
 }
-} // namespace libbig
+
+bool largeInt::operator==(int z)
+{
+    // checks if both inputs have the same sign (positive or negative)
+    if (this->sign != (z >= 0))
+    {
+        return false;
+    }
+    // checks if inputs have a different number of digits
+    // by converting the absolume value of z to a string
+    if (z < 0)
+    {
+        z *= -1;
+    }
+    std::string zStr = std::to_string(z);
```

Comment:
i think this part can be improved a bit, there is no need to convert `z` to `string` we can just directly check it digit by digit
something like
```C++
int N = this->number.length() - 1;
while(z != 0)  {
    if (this->number[N] != (z % 10)) {
         return false;
    }
    z /= 10;
    N--;
}
return true;
```
i hope you get the idea from this sample code snippet.